docs: gpbackup/gprestore clarifications

mkiyama opened the pull request:
-clarified GPDB support (4.3.22.0 and later and 5.5.0 and later)
-clarified how to restore leaf partition data from a backup. Specify the partitioned table.

dyozie commented on gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml:
```diff
@@ -35,9 +35,8 @@
-        systems that support the <codeph>COPY ... ON SEGMENT</codeph> command (Greenplum Database
-        5.1.0 and later<ph otherprops="pivotal">, or 4.3.17.0 and later</ph>).</p>
+      <p>The <codeph>gpbackup</codeph> and <codeph>gprestore</codeph> utilities are supported on
+        Pivotal Greenplum Database 4.3.22.0 and later, Pivotal Greenplum Database 5.5.0 and later. </p>
```
".3.22.0 and later, Pivotal Greenplum" -> ".3.22.0 and later, and on Pivotal Greenplum"

Also, if we're calling out Pivotal releases here the names should be conditionalized accordingly.  Maybe just conditionalize this entire para if it's being framed in terms of "support".

oakbarrett replied:
The intent is to clarify minimum versions of GPDB which support the GA version of gpbackup & gprestore.  Ill defer to Mel & David on how to present this.

dyozie commented on gpdb-doc/dita/admin_guide/managing/backup-gpbackup.xml:
```diff
@@ -395,13 +394,14 @@ public.sales_1_prt_dec17 </codeblock></p><p>Then
-                listed in the file. </p></li>
+                <codeph>gpbackup</codeph>. If you specified leaf partitions in the backup set,
+              specify the partitioned table to restore the leaf partition data. <p>When restoring a
+                backup set that contains data from some leaf partitions of a partitioned tables, the
```
"tables," -> "table,"

mkiyama commented:
updated docs.
-Changed supported information to available for GPDB 5.5.0
-Removed note "not available on SuSE"
-Added information about schemas - schemas not backed up, public schema information.
  Information is a note to table for objects that are backed up and restored.

lisakowen commented:
noticed that the multiple-word options on the gpbackup reference page are prefaced only with a single dash.  isn't it customary to use dash-dash with multiple word options and single dash for single character options?

mkiyama commented:
This is my understanding.
Double dashes was practice for the previous utilities written in Python. It might have been required.
These are written in GO, I believe Go doesn't require the double dashes.

@oakbarrett Please correct me if I am wrong.

oakbarrett commented:
@khuddlefish @chrishajas   please confirm 

Go will handle single and double dashes as the same...      i.e.   --version is the same as -version  and  ---no-compress is the same as -no-compress

lisakowen commented:
if the utility supports both, maybe we should document the dash-dash to be consistent with ?most/all? of the other utilities.

chrishajas commented:
Yes, our flag package interprets both single and double dashes before a flag the same.

dyozie commented:
I that case, I'd like to standardize the docs to use double dashes to be consistent with the rest of the reference material.  @chrishajas - do you know if the command-line help uses double-dashes as well?

chrishajas commented:
@dyozie The CLI help uses single dashes unfortunately and I don't believe there's an easy way to change it. I'm also in favor of using double dashes in all docs for standardization.
